src/main/java/org/openrewrite/java/migrate/lang/IfElseIfConstructToSwitch.java

```diff
@@ -0,0 +1,221 @@
+                    return super.visitSwitch(switch_, ctx);
+                }
+                return super.visitIf(iff, ctx);
+            }
+        });
+    }
+
+    private static class SwitchCandidate {
+        private final Map<J.InstanceOf, Statement> patternMatchers = new LinkedHashMap<>();
+        private @Nullable Expression nullCheckedParameter = null;
+        private @Nullable Statement nullCheckedStatement = null;
+        private @Nullable Statement elze = null;
+
+        @Getter
+        private boolean potentialCandidate = true;
+
+        SwitchCandidate(J.If iff, Cursor cursor) {
+            J.If ifPart = iff;
+            while (potentialCandidate && ifPart != null) {
+                if (ifPart.getIfCondition().getTree() instanceof J.Binary) {
+                    ifPart = handleNullCheck(ifPart, cursor);
+                } else if (ifPart.getIfCondition().getTree() instanceof J.InstanceOf) {
+                    ifPart = handleInstanceOfCheck(ifPart);
+                } else {
+                    noPotentialCandidate();
+                }
+            }
+        }
+
+        private J.@Nullable If handleNullCheck(J.If ifPart, Cursor cursor) {
+            Optional<NullCheck> nullCheck = nullCheck().get(ifPart, cursor);
+            if (nullCheck.isPresent()) {
+                nullCheckedParameter = nullCheck.get().getNullCheckedParameter();
+                nullCheckedStatement = nullCheck.get().whenNull();
+                Statement elsePart = nullCheck.get().whenNotNull();
+                if (elsePart instanceof J.If) {
+                    ifPart = (J.If) elsePart;
+                } else {
+                    elze = elsePart;
+                    ifPart = null;
+                }
+            } else {
+                noPotentialCandidate();
+            }
+            return ifPart;
+        }
+
+        private J.@Nullable If handleInstanceOfCheck(J.If ifPart) {
+            patternMatchers.put((J.InstanceOf) ifPart.getIfCondition().getTree(), ifPart.getThenPart());
+            J.If.Else elsePart = ifPart.getElsePart();
+            if (elsePart != null && elsePart.getBody() instanceof J.If) {
+                ifPart = (J.If) elsePart.getBody();
+            } else {
+                elze = elsePart != null ? elsePart.getBody() : null;
+                ifPart = null;
+            }
+            return ifPart;
+        }
+
+        void noPotentialCandidate() {
+            this.potentialCandidate = false;
+        }
+
+        boolean isValidCandidate() {
+            Expression switchOn = switchOn();
+            // all ifs in the chain must be on the same variable in order to be a candidate for switch pattern matching
+            if (switchOn == null || potentialCandidate && !patternMatchers.keySet().stream()
+                    .map(J.InstanceOf::getExpression)
+                    .allMatch(it -> SemanticallyEqual.areEqual(switchOn, it))) {
+                this.potentialCandidate = false;
```

Comment:
```suggestion
                potentialCandidate = false;
```

Or vice versa, use the `this.` prefix everywhere. At least, be consistent 😉